### PR TITLE
async -> enqueue. await -> enqueueAndWait

### LINKED ISSUE #10: Rename `async` and `await` on both `FIFOQueue` and `ActorQueue`
**Goals**
* Ensure syntax highlighting does not get confused by avoiding the use of language keywords in API

**Non-Goals**
* Avoid a breaking API change

**Discussion**
Our current `async`/`await` API was inspired by both `DispatchQueue`'s `async`/`sync` APIs as well as Swift Concurrency Syntax's `async` and `await` keywords. However, Xcode's syntax highlighting gets confused by this API choice:

screenshot of Xcode syntax highlighting showing `await` as if it were a language keyword rather than a method call

As such, I'd like to change the name of these APIs.

@ktoso's [discussion of creating ordered tasks from nonisolated contexts](https://forums.swift.org/t/task-is-order-of-task-execution-deterministic/51553/31) has me thinking of two possibilities for replacing the `async` and `await` APIs in `FIFOQueue` and `ActorQueue`:

1. `enqueue` and `enqueueAndWait`
2. `send` and `sendAndWait`

I'm currently partial to the former, but am curious to get input on these options. I'm also open to alternative API suggestions!

I hope to include this name change in the `0.2.0` release of this library, which should also include the API changes in #6 and #9.

### PR DESCRIPTION
**Pull request by dfed**
Resolves #10

### CODE REVIEW

**Review by Gabardone — APPROVED** [review]
I like this naming better. The prior version had too much baggage from the other APIs that use the same names.

File: Sources/AsyncQueue/ActorQueue.swift [hunk 1]
```diff
@@ -78,7 +78,7 @@ public final class ActorQueue<ActorType: Actor> {
 
     // MARK: Public
 
-    /// Sets the actor context within which each `async` and `await`ed task will execute.
+    /// Sets the actor context within which each `enqueue` and `enqueueAndWait` task will execute.
```
Comment (bachand) [inline Sources/AsyncQueue/ActorQueue.swift:81]: "`enqueue` and `enqueueAndWait`" -> "`enqueue`ed and `enqueueAndWait`ed"?

File: Sources/AsyncQueue/FIFOQueue.swift [hunk 2]
```diff
@@ -54,7 +54,7 @@ public final class FIFOQueue: Sendable {
     /// Schedules an asynchronous task for execution and immediately returns.
     /// The scheduled task will not execute until all prior tasks – including suspended tasks – have completed.
     /// - Parameter task: The task to enqueue.
-    public func async(_ task: @escaping @Sendable () async -> Void) {
+    public func enqueue(_ task: @escaping @Sendable () async -> Void) {
```
Comment (bachand) [inline Sources/AsyncQueue/FIFOQueue.swift:57]: Do we need to edit this `await` in this line? https://github.com/dfed/swift-async-queue/blob/3454ea6497aeb917b6fe084efd69d95d9c7cff35/Sources/AsyncQueue/FIFOQueue.swift#L25

File: Tests/AsyncQueueTests/FIFOQueueTests.swift [hunk 3]
```diff
@@ -36,46 +36,46 @@ final class FIFOQueueTests: XCTestCase {
 
     // MARK: Behavior Tests
```
Comment (bachand) [inline Tests/AsyncQueueTests/FIFOQueueTests.swift:37]: There are at least a couple places in the file like this that we should clean up.
https://github.com/dfed/swift-async-queue/blob/3454ea6497aeb917b6fe084efd69d95d9c7cff35/Tests/AsyncQueueTests/FIFOQueueTests.swift#L81

**Review by bachand — COMMENTED** [review]
Nice. I agree it's good to pick unique names that don't overlap with Swift language names.

File: Sources/AsyncQueue/FIFOQueue.swift (hunk 2 above)
Comment (dfed) [reply Sources/AsyncQueue/FIFOQueue.swift:57]: oooooh solid find!

**Comment by dfed** [comment]
Pushed feedback up into #13

**Pull request merged**